.evg.yml

```diff
@@ -320,6 +322,104 @@ tasks:
           content_type: application/octet-stream
           permissions: public-read
 
+  - name: release_tag
+    depends_on:
+      - name: lint
+      - name: test_unit
+      - name: test_with_cloud
+      - name: test_e2e
+      - name: build_publish_clis
+    patchable: false
+    commands:
+      - func: "setup_node"
+      - func: "setup_project"
+      - command: s3.get
+        params:
+          aws_key: ${artifacts_aws_key}
+          aws_secret: ${artifacts_aws_secret}
+          remote_file: versions/cloud-prod/CURRENT
+          bucket: realm-clis
+          local_file: manifest.json
+      - command: shell.exec
+        params:
+          shell: "bash"
+          script: |
+            set -e
+            cd realm-cli
+            if [ "${branch_name}" != "master" ]; then
+              echo "skipping release: not on master branch"
+              exit 0
+            fi
+            if [ $(git tag -l "v${cli_version}") ]; then
+              echo "skipping release: tag already exists"
+              exit 0
+            fi
+            touch release.true
+      - command: shell.exec
+        params:
+          shell: "bash"
+          script: |
+            set -e
+            if [ ! -f realm-cli/release.true ]; then
+              # not a release build
+              exit 0
+            fi
+            echo "updating manifest"
+            ./realm-cli/contrib/manifest.js ${build_id} ${cli_version} manifest.json
+      - command: s3.put
+        params:
+          aws_key: ${artifacts_aws_key}
+          aws_secret: ${artifacts_aws_secret}
+          local_file: manifest.json
+          remote_file: versions/cloud-prod/CURRENT
+          bucket: realm-clis
+          content_type: application/octet-stream
+          permissions: public-read
+      - command: shell.exec
+        params:
+          shell: "bash"
+          script: |
+            set -e
+            cd realm-cli
+            if [ ! -f release.true ]; then
+              # not a release build
+              exit 0
+            fi
```

Comment:
we probably can add this above instead of splitting across commands (and thus remove the release.true)

Reply:
I think I'm fine with moving the "tag stuff" above, so that this runs

```yaml
echo "tagging release: v${cli_version}"
PREVIOUS_VERSION=$(git describe --tags $(git rev-list --tags --max-count=1))
BODY=$(git log --no-merges $PREVIOUS_VERSION..HEAD --pretty="format:%s (%h); %an" | sed '$d')
git tag -a -m "Changelog since $PREVIOUS_VERSION:" -m "$BODY" "v${cli_version}"
git push origin master --tags
```

before pushing back to s3 with the potentially updated manifest.

however, the `release.true` could still be helpful by keeping the `npm` stuff separate/after the push to s3.  i'm assuming the `npm publish` part is prone to failure, so trying to keep that at the tail end of this task so that should it fail, we're still in a good place for someone to just locally run `npm publish` without having to do anything else (i.e. `s3` is in a good place and github has its expected tags)